app/src/main/java/com/sun/note_01/data/source/remote/sendjson/SendJson.kt

```diff
@@ -0,0 +1,81 @@
+package com.sun.note_01.data.source.remote.sendjson
+
+import android.os.AsyncTask
+import com.sun.note_01.data.source.remote.OnSendDataJsonListener
+import com.sun.note_01.untils.Constant
+import org.json.JSONObject
+import java.io.*
+import java.net.HttpURLConnection
+import java.net.URL
+
+@Suppress("DEPRECATION", "UNCHECKED_CAST")
+class SendJson<T>(private val listener: OnSendDataJsonListener<T>) :
+    AsyncTask<String?, Void?, String>() {
+
+    private var exception: Exception? = null
+
+    override fun doInBackground(vararg params: String?): String {
+        var dataResponse = ""
+        try {
+            val url = URL(params[POSITION_URL])
+            val email = params[POSITION_EMAIL]
+            val password = params[POSITION_PASSWORD]
+            val urlConnection = url.openConnection() as HttpURLConnection
+            urlConnection.apply {
+                readTimeout = TIME_OUT
+                requestMethod = params[POSITION_REQUEST_METHOD]
+                setRequestProperty("Content-Type", "application/json; utf-8")
+                doOutput = true
+                this.connect()
+            }
+            val jsonObject = ParseDataToJson().createJson(email!!, password!!)
+            setPostRequestContent(urlConnection, jsonObject)
+            if (urlConnection.responseCode == HttpURLConnection.HTTP_OK) {
+                val stream = BufferedInputStream(urlConnection.inputStream)
+                dataResponse = readStream(stream)
+                return dataResponse
+            }
+        } catch (e: Exception) {
+            exception = e
+        }
+        return dataResponse
+    }
+
+    private fun readStream(stream: BufferedInputStream): String {
+        val bufferedReader = BufferedReader(InputStreamReader(stream))
+        val stringBuilder = StringBuilder()
+        bufferedReader.forEachLine { str -> stringBuilder.append(str) }
+        return stringBuilder.toString()
+    }
+
+    override fun onPostExecute(dataResponse: String) {
+        super.onPostExecute(dataResponse)
+        if (dataResponse.isEmpty()) {
+            listener.onError(exception)
+        } else {
+            when (dataResponse.toInt()) {
+                Constant.RESPONSE_FAILED -> {
+                    listener.onFailed()
+                }
+                else -> listener.onSuccess(dataResponse.toInt() as T)
+            }
+        }
```

Comment:
Có cái mình không hiểu được, sao có thể tùy tiện ép kiểu Int thành T được chứ ? 
```suggestion
        when{
            dataResponse.isEmpty() -> listener.onError(exception)
            dataResponse.toInt() == Constant.RESPONSE_FAILED -> listener.onFailed()
            dataResponse.toInt() is T -> listener.onSuccess(dataResponse.toInt())
        }
```